### DISCUSSION TITLE
A detailed guide to making a video TV show with TheMillionaireGame

### DISCUSSION THREAD

**Discussion by ristoryti** [Maerciez/TheMillionaireGame#91] [Guides]
I've spent a lot of time optimizing an easy way to use this program with multiple screens and also cameras for live viewing or recording. This guide is quite detailed but I’ve tried to write it in a way that you don't need a lot of prior technical knowledge to get going.

This guide has two parts. First is for making a recording of the show, and the second one is for a live show. The second one is more complicated. And first, **check my [previous guide](https://github.com/Macronair/TheMillionaireGame/discussions/90) for using dedicated player screens.**

### 1. Recording a TV show

**The simplest setup:**

If you have 2 laptops that you are using as the screens for players, you can also use them for recording. This provides an okay image and relatively good audio and is much easier to set up than the other methods. Try to raise the laptops in some way so that the webcam would be close to the players' eye level, for a better looking image. Check for more framing guidance in the recording tips section.

There are two variants of this setup. The other relies more on editing the video afterwards, while the other needs a bit more time to set up, but you might be able to skip the editing phase.

Option 1
- Use the laptop's built in camera app to record the webcam and internal microphone and leave the recording on in the background. Make a short test recording to see if the audio is recording properly.
- On the third computer, which is running the game, you need a way to record a specific window (the TV screen of the millionairegame), even when it's in the background. OBS is a free software that can do this. Check the editing section below to set up the TV screen properly.
- Use your preferred editing software to edit the three video and audio feeds together. More directions in the editing sections.
- The downside to this that there is no way to see all the time if the recording is still on. Probably it will run without issue but there is always the chance of interruption. And also editing the video is time consuming.

Option 2
- Put both laptops in the same video call, using your preferred video call software, mute the laptop's speakers and put the call in the background.
- On the computer that the game runs on, join the video call, hide self view, make the app always show the person who is speaking, put on full screen and record the call either with the video call app or OBS
- Record millionairegame TV Screen in OBS
- If you record the Video call in OBS, you can overlay the TV screen on top of the video call right away, so you don't need to use any editing software at all. You just need to add some latency to the TV Screen capture, so that it matches the timing of the video call. You need to google how to do this: (OBS how to add delay to video)
- Make a test recording to see if everything works as intended. Adjust the audio levels and video latency if needed.
- The big downside to this method is that if you have an unstable internet connection, or any other issues, the picture might stutter or freeze.
- I haven't tested out this method but it should work. Remember that there is some latency and the video or audio won't work that well if played back live in the same room where the players are playing.
- More obs directions in the live section of this guide

**A more controlled setup:**

**Cameras**
You can use any smartphone or camera for this solution. The device must have enough space to record for longer times, and preferably the option to record longer clips than 30 minutes. You could also just set a timer for yourself to restart the recording after the 30 minutes, if a longer recording isn't possible. You will also need a way to power the camera for a longer duration, a power bank will do just fine.

The simplest way is to just use one camera, that has both of the players in the same picture, and just add the questions overlay afterwards. (I will put some notes about editing the video, later on in the guide.) If your camera can shoot 4k video, you could also crop in to the video to make close up shots afterwards.

If you want a recording that looks much more like the tv show, three cameras is recommended. This means having individual cameras for the contestant and the host, plus a third camera for capturing a wider shot of the room.

You just might have three smartphones available, which still makes this a quite cheap solution. I used an iphone 5s for each camera, and these can be bought as cheap as 10 - 20 $/€ a piece. Android phones will also do just fine.

What could be a bit more challenging, is mounting the cameras. You might need a tripod for each of the cameras, which could be costly, but you might also be able to just put some of the cameras on a shelf or something. If you have some car mounts for smartphones, those might also come in handy. Notice, if you use smartphones, they usually have quite a wide angle lens, which means you need to place them quite close to the players. 

Here is my setup for mounting the smartphones:
- A microphone desk stand with ¼ screw like this: https://www.thomann.de/fi/roadworx_desk_stand.htm
- A dual cold shoe adapter with ¼ screw mount like this:
https://www.ulanzi.com/products/pt-2-extension-plate-ulanzi
- 2x ball heads with cold shoe mounts like this:
https://www.smallrig.com/smallrig-cold-shoe-to-1-4-threaded-adapter-761.html
- 2x Smartphone clamps with ¼ screw holes like this:
https://www.thomann.de/fi/manfrotto_mclamp_smartphone_holder.htm
- 2x cheap usb A to lightning cables
- A power bank with 2 usb ports

The smartphones are mounted to a small desk stand, on the same desk with player monitors. The goal is to have two cameras with easily adjustable angles and approximately at eye level height. This setup costs something around 60 $/€ + the charging equipment, but you might be able to achieve the same results with some cheaper alternatives. This doesn’t include the mount for the third camera. That you might be able to put it just on a shelf or on a table with some books to get a higher angle.

Then, you need a way to record the TV screen. You can use software such as OBS to capture a specific window on your computer. I won't explain this step further, you can find tutorials online. You can also record the sounds of the game using OBS.

**Sound**
This is maybe an even more important part of the recording than placing the cameras. The main idea is that whatever microphone you are using, it's good to place it as close to the players as possible. So, if you don't have any external microphones, it's better to place your cameras close to the players, than far away with a zoom lens.

The best cost effective solution is to hook up each camera with a wired or wireless lavalier microphone. You can get cheap wired lav mics for 5-10 $/€ a piece, and even if they are not the best, if placed close to the speaker, the sound is so much better than any on board microphone if the camera is far away. You could either place the microphone(s) on the table or attach them to the chest of each contestant.
The price of cheap wireless microphones starts maybe from 50/100 $/€ a pair. I happen to have a wireless dji mic pair for other work, so I used that. If you have access to something like that, it's really convenient for this.

Another great solution is to use a stereo audio recorder, that you would place on the table. For example a zoom H1 you could find 30 $/€ second hand.

**Tips on recording**
- It's a good idea to start and end the recording on different devices approximately at the same time. This way you have an easier time syncing up the footage in the editing phase.
- You can also clap your hand a couple of times in front of the camera(s) so that you can use that as a reference point to sync different tracks, especially if you record audio separately.
- It's good general practice to have the cameras on the eye level of the subjects. The wide angle camera can be higher than that.
- It's good to frame the closeup shots in a way that the player's head will be in the top half of the frame, so that the questions overlay won't cover their faces. Check the original show for reference.

**Lighting**
This part is totally optional, but if you really want to make a recording close to the original, this plays an important role. The tv show is quite dimly lit, and the dramatic lights down effect is quite recognizable. The basic idea is to have the table and the players lit with their own light(s), and the rest of the room with another light. Whenever the question is answered, the audience/room light is lit, and with lights down cue, that light turns off.
Simple setup:
1. a desk lamp on the table, or a floor lamp that is close to the players and pointing downwards, is on all the time
2. The room is lit with a normal ceiling light that is turned on and off during the game. You can make this even easier with a cheap smart light that has a remote (10-15$/€)

**Editing**

This can be more or less difficult, depending on your prior editing experience, but I will explain the basic concepts, and encourage you to look online for more help if you are new to video editing.

First, when playing, be sure to change the background color in the game settings to something else than black. Don't use green because the correct answer also flashes green. I used pink in the top right corner of the color selection screen. At least in the current version you need to change this background color EVERY TIME you open the app, so remember to do that.

In your preferred video editing software, for example iMovie on mac or Davinci Resolve (free) on Windows, you need to use a chroma key effect on the captured video from the game. You pick the same color that you previously chose for the TV screen background, and the editing software makes the background transparent. Then you put this video on the top track and the camera footage on tracks below that. Then you need to sync the video clips and audio clips together if you recorded audio separately. If you recorded your audio separately, it might be a good idea to mute the camera's internal audio tracks altogether, and just use the dedicated audio recording.

Then, if you are using multiple camera angles, you need to cut the video. There are good tutorials online on how to edit multi camera footage effectively. Also, listen to the audio levels and see if you need to adjust the game sounds or the microphones, so that they are in a good balance, and that the master volume is not too quiet or too loud.

And last, if you want to add intro or outro graphics, you can find rips of those on online. Now you should be ready to export your final video, MP4/h.264 is usually a good format.

### 2. Using multiple cameras/smartphones for making a "tv show" live viewing experience

In this setup we are using the cameras to show a live feed on a TV or a projector for the audience to watch the game while it's being played. This is the most fun and immersive way to play the game. It's a more complicated setup, but most of the gear you might already have at hand. This can be made to work cheaply, but there are so many factors that affect the good experience, that this gets quite technical. Please also read the previous sections before this one, since many things I've only explained there.

The main challenge in this setup is minimizing the latency. If there is too much delay in the video or audio, the viewing/listening experience is not very good and the video feed might be just more distracting than immersive. What I learned while testing, is that every piece of hardware and software in between the camera lens and the destination screen / projection adds latency. The camera adds latency, the computer adds latency and even your TV adds latency. So if there is a noticeable delay in your end result, there might be many reasons affecting this. I was able to achieve around 180ms / 6 frames (30fps) of video latency which still looks okay. With the cheaper setup, you will probably reach around 8-9 frames, which I will explain later.

On the other hand, if the audience is viewing from another room than the players are in, latency doesn't matter anymore, because we cannot hear or see what is actually happening live. So actually the easiest way to deal with latency is to put the players into a different space, but at the same time you lose some atmosphere and intensity, both for the audience and players. So being in the same room is still nicer. 

**Video**

**Wired methods**

Using two external webcams, with usb extension cables might be the easiest way to setup live video mixing. Used webcams are very cheap and the image quality might be just fine for our purposes. Note that if you are using a high end webcam, long USB 3 extension cables don't work as well as long USB 2 cables. The good thing with this setup is also that webcams often have some built in way to adjust the camera angle, so mounting the camera is easier than with other options. For the third camera, you could have a third webcam, or just use a smartphone as a wireless camera, as explained below.

Some modern DSLR/video cameras also support using the camera as a usb webcam, but test the latency before using this method. Also check if the usb connection provides power to the camera simultaneously.

I will skip using HDMI video mixing method altogether because to do that you would need a hardware video mixer/switcher and also cameras that support clean HDMI video output, and if you already have that kind of equipment, you probably don't need my directions on using it anyway. That being said, I do have an ATEM mini video mixer, but I decided not to use it because it adds some latency, and it's so much hassle to hook up all the wiring for this purpose.

**Wireless methods**

I ended up using old iPhones that send a video feed with NDI over wifi. NDI is a protocol for wireless video over the network with minimal latency. You just need the NDI app for your smartphone and NDI software for your computer. Depending on your video mixing software, you might need to also download a plugin for NDI. On the computer side, everything is free, but the smartphone app costs 10€/$. There are free alternatives but the official NDI HX Camera app is strongly recommended.

Your receiving computer and the smartphones need to be on the same wifi network. When they are connecting for the first time, an internet connection might be needed, but when connecting later on, I've been able to connect just fine without the internet. The wireless router should be placed in a good open position for good transmission.

It's recommended that you only send 720p video from the smartphone, because higher resolutions increase latency. Depending on your phone, the 720p might be the middle, or the leftmost option in the NDI HX app. You need to check if this is the correct resolution from your video software, because the smartphone app doesn't display the resolution. Turning off the microphone from the camera app is also advisable (check the audio section).

Just press the NDI button in the middle and the phone should be streaming.

I don’t really know how many cameras can be streamed on the same network without issues. I have been able to use three cameras, but at some point, if the network gets too crowded, there might be stutters or increases in the latency of all cameras. It’s also a good idea to connect your computer to the wireless router via ethernet for a more stable connection. There are also ethernet adapters available for smartphones, which makes the connection more reliable, but this is totally optional. I’ve stabilized my connections by running the game on one computer and video mixing on another computer, so I can send the monitor signals over a separate network. If you are transmitting the monitor signals on the same network, make sure to decrease the quality of that signal in the monitoring software, so that they don’t use too much of the bandwidth. 

**Audio**

If you are in a small room, I suggest skipping the audio monitoring altogether, and just relying on the player's own voices. If you are in a large room or have a larger audience, some amplification is probably needed. 

There are many ways of capturing audio, as explained in the recording section. However, using the audio from the wireless cameras is not advisable here, because there is so much latency. (Except if you are viewing and listening from a different space, then latency doesn’t matter, as explained before) 200 ms of latency is still bearable with video, but with audio, just over 20 ms of latency starts to get quite noticeable. Therefore, audio capturing is more complicated than in the recording setup. If you are using usb webcams, the latency might be more bearable, but I haven’t tested this out. Also, the webcam’s microphone quality might not be that good. 

Whatever setup you are using, you need to be able to hook up both the microphones and the program sound to your speakers. Using a regular audio mixer for this is good because you can also adjust the audio level for each source easily. If you are routing all of the audio through your computer, you might not need a separate mixer. My setup is a DJI MIC receiver receiving two wireless microphones, straight to my computer, and capturing the program audio in my video software, and outputting all audio from the headphone port to my speakers. This is the easiest setup for audio that I can imagine. The receiver itself has some latency that is still bearable, but maybe only when used with a mac, because the audio latency on mac is much lower than on pc. 

Another cheap-ish but effective way would be to use a cheap dual channel wireless microphone, and an Y-splitter adapter. With the adapter you combine both the microphone signals and the program audio from your computer’s headphone port to one cable that you connect to your speakers. There are three downsides to this:
The volume is reduced, put if your speakers are good, this doesn’t matter
Controlling the output level of the microphones has to be done from the receiver
You are not able to record the microphones directly to your computer. 
If using a wireless microphone, remember that the signal quality might be affected if there is a room full of people. So make sure to place the receiver in a good position. 

Some other options
- If you are using laptops for screens, there might be a low-latency way of routing the laptop’s microphone out of the headphone port.
- Using dynamic xlr microphones with stands
- Using lavalier microphones with extension cables

**Latency**

I did extensive testing with different smartphones and different software. With an iPhone 5s, I was able to achieve a 5 frame latency (not including the streaming software and TV). I didn't test newer iPhones, but it seemed that iPhone 5s provided 1 frame less latency than iPhone 6 and 1,5 frames less than a Galaxy A52. So my general advice is to use old iPhones, but new ones might also work well, I don't know. I also achieved 0,5 frames less latency with a mac computer compared to a PC. Using ethernet adapters for the smartphones didn’t decrease minimum latency, but as the number of wireless signals on the same network increases, this might still prevent some losses in performance.

Unfortunately, on the receiving side, the fastest signal I was able to achieve with a paid software. OBS had 1-2 frames more latency than Resolume, which I was using. I still recommend sticking to OBS, because Resolume is so expensive. I also tested Max, Qlab, livestream studio and touch designer, but none of them matched the performance of Resolume. If you want to get a slight advantage over OBS, you could try Qlab, which is also free, but note that the free version doesn't support audio, as far as I know.

One or two frames of more latency is still okay, but they add up quickly and will start to be too noticeable, and this is why I wanted to explain these effects in detail.

If you want to test how much latency different parts of your system produce, use this setup:
- On youtube, find a 60fps frame counter video and play it on the other side of the screen.
- On the other side of the screen, open ndi video monitor software, connected to your smartphone, sending the video signal. Try to align the both numbers vertically.
- Point your smartphone at the youtube side, so that the ndi monitor displays the numbers running
- You need another smartphone or camera to film the entire screen, preferably recording in 60fps or higher. Film in horizontal mode.
- On your recording, the difference between the numbers is the latency of your wireless signal from your smartphone to the computer
- If you want to know the latency of your video mixing software, replace the ndi video monitor with the output of your video software.
- If you want to know the latency of your TV/Projector, just open youtube on your main display and mirror the screen to your TV/projector, and film both displays.
- If you want to measure the total latency, open youtube in one display and
- Note that 1 frame in 60fps equals to 0,5 frames in 30fps. So if you want to compare to my numbers, just divide your result by two. I listed my numbers in 30fps because it's the final framerate of the show, and 60fps is just more accurate for measuring.

**Using the video mixing software**

These directions might differ a lot depending on your software, but the basic concept is the same. You need to capture the game TV screen to your software, remove the background with chroma key effect, and switch between different camera inputs, while the TV screen overlay stays on top all of the time. 

If you are on obs, in the sources tab, press plus and add a video capture device, and select the ndi camera, or the usb camera if you are using the wired method. (You need to install distroAV plugin first if using ndi) If the video feed doesn’t fill the entire screen, right click the source → Transform → Fit to screen. Then add a window capture source and select the millionairegame TV screen. Then right click the window capture → Filters, in the new window press the plus on the bottom tho add an Effect Filter, select chroma key, change key color type to custom, press select colour, and match the color with the one you picked in the game menu (explained in the editing section of the previous chapter). 

Now your scene is ready. Now, on the scenes list on the left, duplicate this scene and in the duplicate scene, double click the Video capture device, and change the video source to another NDI camera. Now you can switch between the two cameras by changing the scene. Repeat the process for a third camera. There is a scene transition tab open by default: change the mode from fade to cut so it changes instantaneously. 

Then right click on the video display → Full-screen projector (source) and select your TV/ projector. Now you should be ready to go. If you want to add a starting or ending animation, add a new scene and add a Media source, then select your video file. Last, if you want to use your keyboard to switch between cameras, open settings, select hotkeys, locate your scenes and assign a hotkey to Switch to scene, separately with each scene. 

**Some additional tips**

- You probably need to actively adjust the audio levels, so make sure you have a way of doing this while playing
- It’s a good idea to have one person controlling the game and another controlling video/audio
- You can use midi controllers and/or macro recorders to make controlling different programs at the same time easier
- Put all of your wireless gear in places where there is a good line of sight between the transmitter and the receiver to ensure a good signal. Remember that a room full of people might alter the circumstances.
- If the NDI HX app doesn’t start on your old iphone, you can try this method to install an older version of the app. It’s quite technical but it’s the only way AFAIK. The instructions are long but still easy to follow. https://github.com/qnblackcat/How-to-Downgrade-apps-on-AppStore-with-iTunes-and-Charles-Proxy

**Comment by ImmanuelJeswyn** [comment]
Thanks for a detailed guide.

**Comment by Macronair** [comment]
Wow! This is a really great en detailed explanation and guide! Thank you very much for this!
